### PR TITLE
Docs: Add NGINX example for using websockets to Loki

### PR DESCRIPTION
**Pull request by zen**
**What this PR does / why we need it**:

THis PR adds documentation section for Loki, specifically how to configure proxy for connecting to Loki Live tailing feature from NGINX

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

Problems with correct configuration are mentioned here: https://github.com/grafana/grafana/issues/19555
After searching `da web` I couldn't find any working solution specifically for this problem.

### CODE REVIEW

File: docs/sources/features/datasources/loki.md [hunk 1]
```diff
@@ -134,6 +134,35 @@ Note that Live Tailing relies on two Websocket connections: one between the brow
 ```
 ProxyPassMatch "^/(api/datasources/proxy/\d+/loki/api/v1/tail)" "ws://127.0.0.1:3000/$1"
 ```
+Example below shows basic NGINX proxy configuration. Example assumes that Grafana server is available at `http://localhost:3000/`, Loki server is running locally without proxy and your external site uses HTTPS. If you also host Loki behind nginx proxy, you might want to repeat following configuration for Lokis as well.
```
Comment (oddlittlebird) [inline docs/sources/features/datasources/loki.md:137]: Please capitalize NGINX consistently.

Comment (oddlittlebird) [inline docs/sources/features/datasources/loki.md:137]:
```suggestion
Example below shows basic NGINX proxy configuration. The example assumes that the Grafana server is available at `http://localhost:3000/`, Loki server is running locally without proxy, and your external site uses HTTPS. If you also host Loki behind nginx proxy, then you might want to repeat the following configuration for Loki as well.
```

File: docs/sources/features/datasources/loki.md [hunk 2]
```diff
@@ -134,6 +134,35 @@ Note that Live Tailing relies on two Websocket connections: one between the brow
...
+
+In `http` section of NGINX add following map definition:
+```
+  map $http_upgrade $connection_upgrade {
+    default upgrade;
+    '' close;
+  }
+```
+In your `server` section add following configuration:
```
Comment (oddlittlebird) [inline docs/sources/features/datasources/loki.md:146]:
```suggestion
In your `server` section, add following configuration:
```

File: docs/sources/features/datasources/loki.md [hunk 3]
```diff
@@ -134,6 +134,35 @@ Note that Live Tailing relies on two Websocket connections: one between the brow
...
+  location / {
+      proxy_pass          http://localhost:3000/;
+      proxy_set_header    Host              $host;
+      proxy_set_header    X-Real-IP         $remote_addr;
+      proxy_set_header    X-Forwarded-for   $proxy_add_x_forwarded_for;
+      proxy_set_header    X-Forwarded-Proto "https";
+  }
+```
 
 > **Note:** This feature is only available in Grafana v6.3+
```
Comment (oddlittlebird) [inline docs/sources/features/datasources/loki.md:167]:
```suggestion
> **Note:** This feature is only available in Grafana v6.3+.
```

**Comment by oddlittlebird** [comment]
The doc changes generally look all right to me, but the developers need to review them for correctness.

**Comment by marefr** [comment]
My 2 cents. If running loki behind reverse proxy is the problem I would suggest that Loki documentation, https://grafana.com/docs/loki/latest/, should have these nginx configuration examples. This can then be linked from the Grafana loki datasource documentation

**Review by ivanahuckova — COMMENTED** [review]
I really like this and I think this is very useful to have it in documentation, as there are users struggling with this (e.g. https://github.com/grafana/grafana/issues/22905). So as soon as all @oddlittlebird suggestions are committed, I'll approve.

**Comment by marefr** [comment]
Thanks. My bad. Please disregard my comments.

File: docs/sources/features/datasources/loki.md (hunk 2 above)
Comment (zen) [reply docs/sources/features/datasources/loki.md:146]: Are you sure about the comma? What about `the` before `following`

Comment (oddlittlebird) [reply docs/sources/features/datasources/loki.md:146]: Good point. Added.

File: docs/sources/features/datasources/loki.md [hunk 4]
```diff
@@ -134,8 +134,37 @@ Note that Live Tailing relies on two Websocket connections: one between the brow
...
+  location / {
+      proxy_pass          http://localhost:3000/;
+      proxy_set_header    Host              $host;
+      proxy_set_header    X-Real-IP         $remote_addr;
+      proxy_set_header    X-Forwarded-for   $proxy_add_x_forwarded_for;
+      proxy_set_header    X-Forwarded-Proto "https";
+  }
+```
+
+> **Note:** This feature is only available in Grafana v6.3+.
```
Comment (oddlittlebird) [inline docs/sources/features/datasources/loki.md:167]: This sentence needs to be before the feature description, not after.

Comment (zen) [reply docs/sources/features/datasources/loki.md:167]: This is not my change, but looking at the whole file, it's always at the end of documentation snippet.

File: docs/sources/features/datasources/loki.md [hunk 5]
```diff
@@ -134,8 +134,37 @@ Note that Live Tailing relies on two Websocket connections: one between the brow
 ```
 ProxyPassMatch "^/(api/datasources/proxy/\d+/loki/api/v1/tail)" "ws://127.0.0.1:3000/$1"
 ```
+Example below shows basic NGINX proxy configuration. This example assumes that the Grafana server is available at `http://localhost:3000/`, Loki server is running locally without proxy, and your external site uses HTTPS. If you also host Loki behind NGINX proxy, then you might want to repeat the following configuration for Loki as well.
 
-> **Note:** This feature is only available in Grafana v6.3+
+In `http` section of NGINX configuration, add the following map definition:
```
Comment (achatterjee-grafana) [inline docs/sources/features/datasources/loki.md:139]: In the `http` section of NGINX configuration, add the following map definition:

**Review by achatterjee-grafana — COMMENTED** [review]
Asked for a change, otherwise looks OK.

File: docs/sources/features/datasources/loki.md (hunk 5 above)
Comment (zen) [reply docs/sources/features/datasources/loki.md]: Done

File: docs/sources/features/datasources/loki.md (hunk 4 above)
Comment (zen) [reply docs/sources/features/datasources/loki.md:167]: What about this note?

File: docs/sources/features/datasources/loki.md [hunk 6]
```diff
@@ -134,8 +134,37 @@ Note that Live Tailing relies on two Websocket connections: one between the brow
 ```
 ProxyPassMatch "^/(api/datasources/proxy/\d+/loki/api/v1/tail)" "ws://127.0.0.1:3000/$1"
 ```
+Example below shows basic NGINX proxy configuration. This example assumes that the Grafana server is available at `http://localhost:3000/`, Loki server is running locally without proxy, and your external site uses HTTPS. If you also host Loki behind NGINX proxy, then you might want to repeat the following configuration for Loki as well.
```
Comment (achatterjee-grafana) [inline docs/sources/features/datasources/loki.md:137]: The following example below shows the basic NGINX proxy configuration. It assumes that the Grafana server is available at `http://localhost:3000/`, Loki server is running locally without proxy, and your external site uses HTTPS. If you also host Loki behind NGINX proxy, then you might want to repeat the following configuration for Loki as well.

**Review by achatterjee-grafana — COMMENTED** [review]
Added copy-edit suggestion.

**Review by ivanahuckova — APPROVED** [review]
LGTM

**Comment by ivanahuckova** [comment]
@zen documentation looks great. Thanks again for you work! Highly appreciated. As a last thing, could you please resolve the merge issue - it looks like we have moved `loki.md` from `docs/sources/features/datasources/loki.md` to `docs/sources/datasources/loki.md`. As soon as that is resolved, I can merge. 🙂

**Comment by zen** [comment]
> @zen documentation looks great. Thanks again for you work! Highly appreciated. As a last thing, could you please resolve the merge issue - it looks like we have moved `loki.md` from `docs/sources/features/datasources/loki.md` to `docs/sources/datasources/loki.md`. As soon as that is resolved, I can merge. 🙂

I was actually trying to resolve this but got stuck. I'll read contributor docs how to approach that. If that would not halp, can I expect some assistance?

**Comment by ivanahuckova** [comment]
Yeah, absolutely. But I think that the easiest way would be to accept all changes from master, take all of your contribution and add it to the new loki file`docs/sources/datasources/loki.md` and remove all changes in `docs/sources/features/datasources/loki.md` file.

**Comment by zen** [comment]
@ivanahuckova I completely missed location change, now it should be ok I guess. Let me know if there is anything needed from me

**Comment by ivanahuckova** [comment]
Awesome! Thanks again, this docs update is really great and it is going to help a lot of users. Highly appreciated!! 🙏

**Comment by nh2** [comment]
Needs a fix: https://github.com/grafana/grafana/pull/97609

**Pull request merged**